### ISSUE TITLE
get_or_create method

### ISSUE THREAD

**Issue by Bartleby2718** [Azure/azure-cosmos-python#43]
Correct me if there's a better way, but I found myself copying and pasting the following code (or something similar) over and over again :

```
# Given the strings `database_id` and `collection_id` and a relevant `client` (instance of `DocumentClient`)
try:
    database = next(data for data in client.ReadDatabases() if data['id'] == database_id)
except StopIteration:
    database = client.CreateDatabase({'id': database_id})
database_link = database['_self']
try:
    collection = next(data for data in client.ReadCollections(database_link) if data['id'] == collection_id)
except StopIteration:
    collection = client.CreateCollection(database_link, {'id': collection_id})
```

So, how about a `get_or_create_sometype`method in the `DocumentClient` class of pydocumentdb/document_client.py, like [the one in Django QuerySet class](https://github.com/django/django/blob/master/django/db/models/query.py#L463)? Then, the code above would be much shorter and more readable like this:

```
# Given the strings database_id and `collection_id` and a relevant `client` (instance of `DocumentClient`)
database = client.get_or_create_database(database_id)
database_link = database['_self']
collection = client.get_or_create_collection(database_link, collection_id)
```

p.s. [The link to guidelines for contributing to this repository](https://github.com/Azure/azure-documentdb-python/blob/master/Contributing.md) is broken.

**Comment by rnagpal** [comment]
I fixed the link to contribution guidelines. Feel free to submit a PR for this or else we would add it to our backlog and get to it in near future.

**Comment by southpolesteve** [comment]
Closing this issue due to age. Active development of our Python SDK has moved to https://github.com/Azure/azure-sdk-for-python. We have also released 4.0 which includes many updates and new features. If this issue still needs our attention, please reopen it in the new repository.
